docs/source/notes/autograd.rst

```diff
@@ -219,24 +219,44 @@ Autograd for Complex Numbers
 **What notion of complex derivative does PyTorch use?**
 *******************************************************
 
-PyTorch follows `JAX's <https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Complex-numbers-and-differentiation>`_
-convention for autograd for Complex Numbers.
+PyTorch uses the Wirtinger Calculus to compute gradients for functions with complex valued input and (or)
+output. PyTorch's autograd convention for complex numbers allows gradient value to be used in
+Gradient Descent with the same update rule as that in the case of real numbers. This convention works for
+optimization problems where the objective function is real valued.
 
-Suppose we have a function :math:`F: ℂ → ℂ` which we can decompose into functions u and v
-which compute the real and imaginary parts of the function:
+Suppose we have a function :math:`F: ℂ → ℂ`, which is a part of a :math:`G: ℂ → ℝ` function,
+and can be decomposed into functions :math:`u` and :math:`v` which compute the real and imaginary parts of the function:
 
     .. code::
 
         def F(z):
             x, y = real(z), imag(z)
-            return u(x, y) + v(x, y) * 1j
+            s = u(x, y) + v(x, y) * 1j
+            return s
 
-where :math:`1j` is a unit imaginary number.
+where :math:`1j` is a unit imaginary number and :math:`u` and :math:`v` are :math:`ℝ^{2} → ℝ` functions.
 
-We define the :math:`JVP` for function :math:`F` at :math:`(x, y)` applied to a tangent
-vector :math:`c+dj \in C` as:
+We define the Vector-Jacobian Product `VJP` of :math:`F` at :math:`x + yj` for grad output vector :math:`grad\_out` as:
```

Comment:
As a complex numbers n00b, I think this documentation is a great step to helping developers understand how to write (or update) gradient formulas to support complex numbers.

I think something else that would be helpful is if there was an Appendix somewhere that stepped through the derivation of some operator (like torch.mul, or torch.dot). When I started out understanding how complex number gradients worked, I tried to derive some formulas (e.g., for torch.mul) and compare them against the existing gradient formulas.

Perhaps that would be better for developer documentation though -- I'm not sure if the purpose of notes/autograd is for users to look at and understand what's going on, or for developers to read.

Reply:
@zou3519 thanks for the feedback! I added the screenshots in the description as per your suggestion. I agree that it would be useful to step through the derivation for one of the operators. However, I am not sure if this doc note is the right place to do that. We can possibly include a gradient derivation for one of the operators here: https://pytorch.org/docs/stable/complex_numbers.html#autograd along with an example of how the gradients can be used in an optimization problem.